utils/src/ooinstall/cli_installer.py

```diff
@@ -576,29 +594,32 @@ def get_missing_info_from_user(oo_cfg):
     confirm_continue(message)
     click.clear()
 
-    if oo_cfg.settings.get('ansible_ssh_user', '') == '':
-        oo_cfg.settings['ansible_ssh_user'] = get_ansible_ssh_user()
+    if not oo_cfg.deployment.variables.get('ansible_ssh_user', ''):
+        oo_cfg.deployment.variables['ansible_ssh_user'] = get_ansible_ssh_user()
         click.clear()
 
-    if oo_cfg.settings.get('variant', '') == '':
+    if not oo_cfg.deployment.variables.get('variant', ''):
         variant, version = get_variant_and_version()
-        oo_cfg.settings['variant'] = variant.name
-        oo_cfg.settings['variant_version'] = version.name
+        oo_cfg.deployment.variables['variant'] = variant.name
+        oo_cfg.deployment.variables['variant_version'] = version.name
```

Comment:
variant and variant_version should not be attributes of the deployment. In the example config they are top level attributes of the config file since they drive the installer workflow and not deployment configuration.